stdlib/public/core/RangeSet.swift

```diff
@@ -0,0 +1,376 @@
+    for range in other._ranges {
+      insert(contentsOf: range)
+    }
+  }
+  
+  /// Removes the contents of this range set that aren't also in the given
+  /// range set.
+  ///
+  /// - Parameter other: A range set to intersect with.
+  public mutating func formIntersection(_ other: RangeSet<Bound>) {
+    self = self.intersection(other)
+  }
+  
+  /// Removes the contents of this range set that are also in the given set
+  /// and adds the contents of the given set that are not already in this
+  /// range set.
+  ///
+  /// - Parameter other: A range set to perform a symmetric difference against.
+  public mutating func formSymmetricDifference(
+    _ other: __owned RangeSet<Bound>
+  ) {
+    self = self.symmetricDifference(other)
+  }
+  
+  /// Removes the contents of the given range set from this range set.
+  ///
+  /// - Parameter other: A range set to subtract from this one.
+  public mutating func subtract(_ other: RangeSet<Bound>) {
+    for range in other._ranges {
+      remove(contentsOf: range)
+    }
+  }
+  
+  /// Returns a new range set containing the contents of both this set and the
+  /// given set.
+  ///
+  /// - Parameter other: The range set to merge with this one.
+  /// - Returns: A new range set.
+  public __consuming func union(
+    _ other: __owned RangeSet<Bound>
+  ) -> RangeSet<Bound> {
+    var result = self
+    result.formUnion(other)
+    return result
+  }
+  
+  /// Returns a new range set containing the contents of both this set and the
+  /// given set.
+  ///
+  /// - Parameter other: The range set to merge with this one.
+  /// - Returns: A new range set.
+  public __consuming func intersection(
+    _ other: RangeSet<Bound>
+  ) -> RangeSet<Bound> {
+    return RangeSet(_ranges: _ranges._intersection(other._ranges))
+  }
+  
+  /// Returns a new range set representing the values in this range set or the
+  /// given range set, but not both.
+  ///
+  /// - Parameter other: The range set to find a symmetric difference with.
+  /// - Returns: A new range set.
+  public __consuming func symmetricDifference(
+    _ other: __owned RangeSet<Bound>
+  ) -> RangeSet<Bound> {
+    return union(other).subtracting(intersection(other))
+  }
+  
+  /// Returns a new set containing the contents of this range set that are not
+  /// also in the given range set.
+  ///
+  /// - Parameter other: The range set to subtract.
+  /// - Returns: A new range set.
+  public func subtracting(_ other: RangeSet<Bound>) -> RangeSet<Bound> {
+    var result = self
+    result.subtract(other)
+    return result
+  }
+  
+  /// Returns a Boolean value that indicates whether this range set is a
+  /// subset of the given set.
+  ///
+  /// - Parameter other: A range set to compare against.
+  /// - Returns: `true` if this range set is a subset of `other`;
+  ///   otherwise, `false`.
+  public func isSubset(of other: RangeSet<Bound>) -> Bool {
+    self.intersection(other) == self
```

Comment:
Nit: This is quite a bit more expensive than it needs to be -- it can be done by a single read-only iteration over both sets. We can definitely leave optimizing it for later, though. (Especially given we don't have many tests.)

Reply:
I went ahead and rewrote it + added a test for it since it wasn't too bad